### PR TITLE
Add a planner for ECS deployment

### PR DESCRIPTION
**Pull request by TaKO8Ki**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Ref #1568

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/planner/ecs/ecs.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		return
+	}
+
+	// When no pipeline was configured, perform the quick sync.
+	if cfg.Pipeline == nil || len(cfg.Pipeline.Stages) == 0 {
+		out.Stages = buildQuickSyncPipeline(cfg.Input.AutoRollback, time.Now())
+		out.Summary = fmt.Sprintf("Quick sync to deploy image %s and configure all traffic to it (pipeline was not configured)", out.Version)
+		return
+	}
+
```
Comment (nghialv) [inline pkg/app/piped/planner/ecs/ecs.go:88]: Configure quick sync in this case too. 👍

Comment (TaKO8Ki) [reply pkg/app/piped/planner/ecs/ecs.go:88]: Done with it!

File: pkg/app/piped/planner/ecs/pipeline_test.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+			wantAutoRollback: true,
+		},
+		{
+			name:             "don't want auto rollback stage",
+			wantAutoRollback: true,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			gotStages := buildQuickSyncPipeline(tc.wantAutoRollback, time.Now())
```
Comment (nghialv) [inline pkg/app/piped/planner/ecs/pipeline_test.go:28]: nit: `stages`

File: pkg/app/piped/planner/ecs/pipeline_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			var gotAutoRollback bool
```
Comment (nghialv) [inline pkg/app/piped/planner/ecs/pipeline_test.go:29]: nit: `autoRollback`

**Comment by nghialv** [comment]
Thank you.
/lgtm

File: pkg/app/piped/planner/ecs/pipeline_test.go (hunk 2 above)
Comment (TaKO8Ki) [reply pkg/app/piped/planner/ecs/pipeline_test.go]: Done with it!

**Comment by khanhtc1202** [comment]
🚀 
/approve

**Pull request merged**